File: Directory/53. Maximum Subarray/Solution.cpp
/*
* @Author: yufeng
* @Date:   2018-10-08 16:23:29
* @Last Modified by:   yufeng
* @Last Modified time: 2018-10-08 16:27:17
*/
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=0,ans=nums[0];
        for(auto x: nums){
        	sum+=x;
        	if(ans<sum){
        		ans=sum;
        	}
        	if(sum<0){
        		sum=0;
        	}
        }
        return ans;
    }
};

File: Directory/53. Maximum Subarray/Solution.py
# -*- coding: utf-8 -*-
# @Author: yufeng
# @Date:   2018-10-14 09:04:49
# @Last Modified by:   yufeng
# @Last Modified time: 2018-10-14 09:07:05
# @Email: fzhiy270@sina.com
# @Blog: http://fzhiy.com/

class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sum,ans = 0,nums[0]
        for num in nums:
        	sum+=num
        	ans=max(ans, sum)
        	sum=max(sum, 0)
        return ans